Makefile

```diff
@@ -1,22 +1,92 @@
+#! /usr/bin/make -f
+
+# Project variables.
+VERSION := development
+PROJECT_NAME := $(shell basename "$(PWD)")
 DATE := $(shell date '+%Y-%m-%dT%H:%M:%S')
-VERSION = development
 HEAD = $(shell git rev-parse HEAD)
 LD_FLAGS = -X github.com/tendermint/starport/starport/internal/version.Version='$(VERSION)' \
 	-X github.com/tendermint/starport/starport/internal/version.Head='$(HEAD)' \
 	-X github.com/tendermint/starport/starport/internal/version.Date='$(DATE)'
 BUILD_FLAGS = -mod=readonly -ldflags='$(LD_FLAGS)'
 
+# Go related variables.
+GOBASE := $(shell pwd)
+GOBIN := $(GOBASE)/dist
+GOCMD := $(GOBASE)/starport/cmd/starport
+GOCILINT := $(GOPATH)/bin/golangci-lint
+GORELEASER := $(GOPATH)/bin/goreleaser
+STARPORT := $(GOPATH)/bin/starport
+
+# For go 1.13-1.15 compatibility
+# https://maelvls.dev/go111module-everywhere/
+export GO111MODULE=on
+
+## install: Install de binary.
 install:
 	@echo Installing Starport...
 	@go install $(BUILD_FLAGS) ./...
 	@starport version
 
+## build-binary: Build the binary.
+build-binary: clean
+	@echo Building Starport...
+	@go build $(BUILD_FLAGS) -o $(GOBIN)/$(PROJECT_NAME) $(GOCMD)
+
+$(STARPORT): install
+
+$(GORELEASER):
+	@echo Installing goreleaser...
+	@go install github.com/goreleaser/goreleaser@latest
+
+## goreleaser: Build the binaries with Goreleaser.
+goreleaser: clean $(GORELEASER)
```

Comment:
We use goreleaser in the CI to automate the process of releasing binaries, any use case of having it here as well?

Reply:
It's easier for the CI/CD call and generate the all binaries version, but no usage now, removing...

Reply:
local testing of stuff that runs in ci is the reason to keep it, I reckon.

Reply:
> What if we kept build-binary and had it output into a folder called `build`?
> 
> It was nice to not have to wonder which version is on $PATH

Yeah, I will put back the changes, but let's use the `dist` folder to use the same folder as `goreleaser`